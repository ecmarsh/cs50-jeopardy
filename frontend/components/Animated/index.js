import Ease from './_Ease';
import Fade from './_Fade';
import Grow from './_Grow';
import Loud from './_Loud';
import Shift from './_Shift';
import Slide from './_Slide';
import Tally from './_Tally';
import Tree from './_Tree';
import Unfocus from './_Unfocus';

export { Ease, Fade, Grow, Loud, Shift, Slide, Tally, Tree, Unfocus };
