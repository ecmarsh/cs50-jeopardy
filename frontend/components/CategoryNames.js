import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import * as R from 'ramda';
import formatRound from '../lib/formatRound';
import Error from './ErrorMessage';
import Form from './styles/Form';
import Button from './styles/Button';
import { TextMuted, Divider } from './styles/Utilities';

const UPDATE_CATEGORY_MUTATION = gql`
  mutation UPDATE_CATEGORY_MUTATION($categoryId: ID!, $categoryName: String) {
    updateCategory(id: $categoryId, name: $categoryName) {
      id
      name
    }
  }
`;

class CategoryNames extends Component {
  static propTypes = {
    categories: PropTypes.array.isRequired,
  };

  state = {
    categories: [],
    called: false,
  };

  componentDidMount() {
    const { categories } = this.props;
    this.setState({ categories });
  }

  handleChange = e => {
    e.preventDefault();
    const { id, value } = e.target;
    const categories = [...this.state.categories];
    const idx = R.findIndex(R.propEq('id', id))(categories);
    categories[idx].name = value;
    categories[idx].updated = true; // flag to include in mutation
    this.setState({ categories });
  };

  handleSubmit = (e, updateCategoryMutation) => {
    // Stop default form submission
    e.preventDefault();
    // Get list of changed categories to update
    const updatedCategories = R.filter(R.has('updated'), this.state.categories);
    // Update category mutation
    const doMutation = async category => {
      const { id, name } = category;
      const res = await updateCategoryMutation({
        variables: { categoryId: id, categoryName: name },
      });
    };
    // Invoke mutation on each updated category
    R.forEach(doMutation, updatedCategories);
    // Called to flash saved message
    this.setState({ called: true });

    setTimeout(this.flashUpdatedConfirmation, 2000);
  };

  flashUpdatedConfirmation = () => {
    this.setState({ called: false });
  };

  render() {
    const { categories, called } = this.state;
    return (
      <Mutation mutation={UPDATE_CATEGORY_MUTATION}>
        {(updateCategory, { loading, error }) => {
          return (
            <Form
              data-test="category-names-form"
              className="noBg"
              onSubmit={e => this.handleSubmit(e, updateCategory)}
            >
              <h2>Category Names</h2>
              <Error error={error} />
              <fieldset disabled={loading} aria-busy={loading}>
                {categories.map((category, i) => (
                  <div key={category.id}>
                    {i === 0 && <h4>{formatRound(1)}</h4>}
                    {i === 5 && <h4>{formatRound(2)}</h4>}
                    {i === 10 && <h4>{formatRound(3)}</h4>}
                    <label
                      htmlFor={`category-${i}`}
                      style={{
                        flexDirection: 'row',
                        width: '100%',
                      }}
                    >
                      <TextMuted
                        style={{ textAlign: 'center', flexBasis: '25%' }}
                      >
                        CATEGORY {i + 1}:
                      </TextMuted>
                      <input
                        type="text"
                        name={`category-${i}`}
                        id={category.id}
                        value={category.name}
                        onChange={e => this.handleChange(e)}
                        style={{
                          flexBasis: '60%',
                          width: 'auto',
                        }}
                      />
                    </label>
                  </div>
                ))}
                <div style={{ margin: '0 auto' }}>
                  <TextMuted
                    data-test="category-names-save-confirmation"
                    hidden={!called}
                  >
                    Category Names Saved!
                  </TextMuted>
                  <Button
                    primary
                    type="submit"
                    style={{ margin: '1rem auto 2rem' }}
                  >
                    Save Category Names
                  </Button>
                  <Divider />
                  {this.props.children}
                </div>
              </fieldset>
            </Form>
          );
        }}
      </Mutation>
    );
  }
}

export default CategoryNames;
export { UPDATE_CATEGORY_MUTATION };
