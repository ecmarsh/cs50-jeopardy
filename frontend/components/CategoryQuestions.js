import React from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import AddToCategory from './CategoryQuestionAdd';
import { categoriesPerPage } from '../config';

const CATEGORY_QUESTIONS_QUERY = gql`
  query CATEGORY_QUESTIONS_QUERY($categoryId: ID) {
    categoryQuestions(where: { category: { id: $categoryId } }) {
      id
      difficulty
      isDouble
      answered
      question {
        id
        question
        answer
      }
      category {
        id
        name
      }
    }
  }
`;

const CategoryQuestions = ({ gameName, categoryId, round }) => (
  <Query query={CATEGORY_QUESTIONS_QUERY} variables={{ categoryId }}>
    {({ data, error }) => {
      if (error) return <p>Error...</p>;
      if (data.categoryQuestions)
        return (
          <>
            {data.categoryQuestions.map((categoryQuestion, index) => (
              <AddToCategory
                key={categoryQuestion.id}
                gameName={gameName}
                categoryId={categoryId}
                categoryName={categoryQuestion.category.name}
                categoryQuestionId={categoryQuestion.id}
                difficulty={categoryQuestion.difficulty}
                question={categoryQuestion.question}
                loaded={categoryQuestion.question !== null}
                skip={categoriesPerPage * (round - 1)}
                cacheCategoryIndex={index}
              />
            ))}
          </>
        );
    }}
  </Query>
);

export default CategoryQuestions;
export { CATEGORY_QUESTIONS_QUERY };
