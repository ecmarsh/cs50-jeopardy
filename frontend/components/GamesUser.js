import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import Link from 'next/link';
import { format } from 'date-fns';
import { GAME_SUMMARY_QUERY } from './GameSummary';
import DeleteGame from './GameDelete';
import Error from './ErrorMessage';
import LeftColumn from './styles/List';
import StyledListItem from './styles/ListItem';
import { CloseSquareO } from './styles/icons';
import { TextOnOff } from './styles/Utilities';

const USER_GAMES_QUERY = gql`
  query USER_GAMES_QUERY {
    myGames {
      id
      name
      createdAt
      user {
        id
        name
      }
      questions {
        id
      }
    }
  }
`;

const GameStatus = ({ gameName }) => (
  <Query query={GAME_SUMMARY_QUERY} variables={{ gameName }}>
    {({ data: { categoryQuestionsConnection }, loading, error }) => {
      if (loading)
        return (
          <p>
            <small>Loading...</small>
          </p>
        );
      if (error)
        return (
          <p>
            <small>Questions Status</small>
          </p>
        );
      const loadedQuestions = categoryQuestionsConnection.aggregate.count;
      return (
        <p>
          <small>
            Question Status
            <TextOnOff textMuted={loadedQuestions < 51}>
              {loadedQuestions} / 51
            </TextOnOff>
          </small>
        </p>
      );
    }}
  </Query>
);
GameStatus.propTypes = {
  gameName: PropTypes.string.isRequired,
};

const UserGameListItem = ({ game: { id, name, createdAt } }) => (
  <StyledListItem key={id}>
    <Link
      href={{
        pathname: `/setup`,
        query: { game: name },
      }}
    >
      <a>
        <h4>{name}</h4>
        <GameStatus gameName={name} />
        <p>
          <small>{format(createdAt, 'MMMM DD, YYYY h:mm a')}</small>
        </p>
      </a>
    </Link>
    <DeleteGame gameName={name}>
      <CloseSquareO />
    </DeleteGame>
  </StyledListItem>
);
UserGameListItem.propTypes = {
  game: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    createdAt: PropTypes.string,
  }).isRequired,
};

const Heading = ({ isSolo }) => {
  function HeaderInner() {
    return (
      <Fragment>
        <div className="heading">
          <h3>My Games</h3>
          <Link href="/start">
            <a>+ Create New Game</a>
          </Link>
          <hr />
        </div>
      </Fragment>
    );
  }
  if (isSolo) {
    return (
      <LeftColumn>
        <HeaderInner />
      </LeftColumn>
    );
  }
  return <HeaderInner />;
};
Heading.propTypes = {
  isSolo: PropTypes.bool,
};

const UserGamesList = () => (
  <Query query={USER_GAMES_QUERY}>
    {({ data: { myGames }, loading, error }) => {
      if (loading) return 'Loading...';
      if (!myGames) return <Heading isSolo />;
      return (
        <LeftColumn>
          <Heading />
          <Error error={error} />
          <ul>
            {myGames.map(game => (
              <UserGameListItem key={game.id} game={game} />
            ))}
          </ul>
        </LeftColumn>
      );
    }}
  </Query>
);

export default UserGamesList;
export { USER_GAMES_QUERY, UserGameListItem, GameStatus };
