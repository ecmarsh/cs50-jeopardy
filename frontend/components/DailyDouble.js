import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import SoundEffect from './SoundEffect';
import Button from './styles/Button';
import formatScore from '../lib/formatScore';
import { Ease, Loud } from './Animated';

function DailyDouble({ minWager, maxWager, setQuestionPrice, isChosenTeam }) {
  const [wager, setWager] = useState(minWager);
  return (
    <div>
      <SoundEffect sound="daily-double" />
      <Loud off={isChosenTeam}>Daily Double!</Loud>
      {isChosenTeam && (
        <Ease
          data-test="daily-double-form"
          onSubmit={e => {
            e.preventDefault();
            setQuestionPrice(wager);
          }}
        >
          <Field>
            <label>
              <p>
                Set team wager from {formatScore(minWager)} -{' '}
                {formatScore(maxWager)}
              </p>
              <input
                type="number"
                step={100}
                min={minWager}
                max={maxWager}
                value={wager}
                onChange={e => setWager(e.target.value)}
              />
            </label>
            <Button primary type="submit">
              Set Wager
            </Button>
            <Button
              tertiary
              className="contrast"
              type="button"
              onClick={() => setWager(maxWager)}
            >
              Make it a True Daily Double!
            </Button>
          </Field>
        </Ease>
      )}
    </div>
  );
}
DailyDouble.propTypes = {
  minWager: PropTypes.number.isRequired,
  maxWager: PropTypes.number.isRequired,
  setQuestionPrice: PropTypes.func.isRequired,
  isChosenTeam: PropTypes.oneOfType([PropTypes.bool, PropTypes.string])
    .isRequired,
};
DailyDouble.defaultProps = {
  minWager: 0,
  maxWager: 1000,
  setQuestionPrice: () => console.log('Missing price setter prop fn'),
  isChosenTeam: false,
};

const Field = styled.fieldset`
  border: none;
  border-image: none;
  line-height: 2;
  label {
    font-size: 2.4rem;
    line-height: 1.5;
  }

  input {
    padding: 1.2rem;
    font-size: 1.6rem;
    text-align: center;
    margin: 2rem auto;
    width: 10rem;
    box-shadow: 0 0 1rem rgba(255, 255, 255, 0.5);
    color: ${props => props.theme.darkGrey};

    &:focus {
      outline-color: ${props => props.theme.primaryDark};
    }
  }
  button {
    display: block;
    margin: 1rem auto;
    &.contrast {
      color: white;
      opacity: 0.8;
      margin-top: 1rem;
      &:hover {
        color: white;
        opacity: 1;
      }
    }
  }
`;

export default DailyDouble;
