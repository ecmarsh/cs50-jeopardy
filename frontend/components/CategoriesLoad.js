import React, { Component } from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import { categoriesPerPage } from '../config';
import CategoryQuestions from './CategoryQuestions';
import { Heading } from './Views';
import Pagination from './Pagination';
import Error from './ErrorMessage';
import Main from './styles/Main';
import { Divider } from './styles/Utilities';

const CATEGORIES_QUERY = gql`
  query CATEGORIES_QUERY($gameName: String, $first: Int, $skip: Int) {
    categories(
      where: { game: { name: $gameName } }
      first: $first
      skip: $skip
    ) {
      id
      name
    }
  }
`;

class LoadCategories extends Component {
  state = {};

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    const { game, round, headingProps } = this.props;
    return (
      <Query
        query={CATEGORIES_QUERY}
        variables={{
          gameName: game,
          first: categoriesPerPage,
          skip: categoriesPerPage * (round - 1),
        }}
      >
        {({ data, loading, error }) => {
          if (loading) return <p>Loading...</p>;
          if (error) return <Error error={error} />;
          if (!data.categories) {
            return <Heading game={game} {...headingProps} />;
          }
          return (
            <Main>
              <Heading game={game} {...headingProps} />
              <Divider style={{ margin: '0 auto' }} />
              <Pagination gameName={game} round={round} />
              <div className="canvas">
                {data.categories.map(category => (
                  <div
                    key={'category' + category.id}
                    className="column-wrapper"
                  >
                    <ul>
                      <li>{category.name}</li>
                      <CategoryQuestions
                        categoryId={category.id}
                        gameName={game}
                        round={round}
                      />
                    </ul>
                  </div>
                ))}
              </div>
            </Main>
          );
        }}
      </Query>
    );
  }
}

export default LoadCategories;
export { CATEGORIES_QUERY };
