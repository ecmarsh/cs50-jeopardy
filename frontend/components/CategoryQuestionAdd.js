import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import DropTarget from './styles/DropTarget';
import Error from './ErrorMessage';
import RemoveFromCategory from './CategoryQuestionRemove';
import { GAME_QUESTIONS_QUERY } from './QuestionsList';
import { CATEGORY_QUESTIONS_QUERY } from './CategoryQuestions';
import { GAME_SUMMARY_QUERY } from './GameSummary';
import * as R from 'ramda';

const ADD_TO_CATEGORY_MUTATION = gql`
  mutation ADD_TO_CATEGORY_MUTATION($id: ID!, $questionId: ID!) {
    updateCategoryQuestion(id: $id, questionId: $questionId) {
      id
      question {
        categoryQuestions {
          id
        }
      }
    }
  }
`;

class AddToCategory extends Component {
  static propTypes = {
    loaded: PropTypes.bool,
    question: PropTypes.object,
    categoryQuestionId: PropTypes.string.isRequired,
    categoryId: PropTypes.string.isRequired,
    difficulty: PropTypes.number.isRequired,
    categoryName: PropTypes.string.isRequired,
    gameName: PropTypes.string.isRequired,
  };

  state = {
    questionId: String(),
    question: String(),
  };

  componentDidMount = () => {
    if (this.props.question) {
      const { id, question } = this.props.question;
      this.setState({
        questionId: id,
        question,
      });
    }
  };

  handleDragOver = e => {
    e.preventDefault();
    e.persist(); // don't pool the events
    e.target.style.transform = 'scale(1.2)';
    e.dataTransfer.dropEffect = 'move';
  };

  handleDragLeave = e => {
    e.target.style.transform = 'scale(1)';
  };

  handleDrop = async (e, categoryMutation) => {
    e.persist();
    e.target.style.transform = 'scale(1)';
    e.dataTransfer.dropEffect = 'move';

    const { categoryId, categoryQuestionId, difficulty, gameName } = this.props,
      questionId = e.dataTransfer.getData('id'),
      question = e.dataTransfer.getData('question'),
      answer = e.dataTransfer.getData('answer');

    this.setState({ questionId, question });

    const res = await categoryMutation({
      variables: { id: categoryQuestionId, questionId },
      update: (cache, { data: { updateCategoryQuestion } }) => {
        /* Update category question with added question */
        const categoryQuestionsQuery = {
          query: CATEGORY_QUESTIONS_QUERY,
          variables: { categoryId },
        };
        // Read the data
        const { categoryQuestions } = cache.readQuery(categoryQuestionsQuery);
        // Add the question
        categoryQuestions[difficulty - 1].question = {
          answer,
          id: questionId,
          question,
          __typename: 'Question',
        };
        // Write back the cache
        cache.writeQuery({
          data: { categoryQuestions },
          ...categoryQuestionsQuery,
        });

        /* Update question to know it's in a category now */
        const gameQuestionsQuery = {
          query: GAME_QUESTIONS_QUERY,
          variables: { gameName },
        };
        // Read data to questions variable
        const { questions } = cache.readQuery(gameQuestionsQuery);
        // Add new question into matching question's category list
        const idx = R.findIndex(R.propEq('id', questionId), questions);
        const questionToAdd = {
          id: updateCategoryQuestion.id,
          __typename: 'CategoryQuestion',
          category: {
            id: categoryId,
            __typename: 'Category',
          },
        };
        questions[idx].categoryQuestions.push(questionToAdd);
        // Write back cache with updates
        gameQuestionsQuery.data = { questions };
        //console.log(gameQuestionsQuery);
        cache.writeQuery(gameQuestionsQuery);
      },
    });
  };

  render() {
    const {
      loaded,
      question,
      categoryQuestionId,
      categoryId,
      difficulty,
      categoryName,
      gameName,
    } = this.props;

    return (
      <Mutation
        mutation={ADD_TO_CATEGORY_MUTATION}
        refetchQueries={[
          {
            query: GAME_SUMMARY_QUERY,
            variables: { gameName },
          },
        ]}
      >
        {(updateCategoryQuestion, { loading, error }) => {
          if (loading) return <DropTarget>Adding..</DropTarget>;
          if (error) return <Error error={error} />;
          return (
            <DropTarget
              id={`${categoryName}-${categoryId}`}
              onDragOver={!loaded ? this.handleDragOver : undefined}
              onDragLeave={!loaded ? this.handleDragLeave : undefined}
              onDrop={e => this.handleDrop(e, updateCategoryQuestion)}
              className={loaded ? '--isLoaded' : '--isNotLoaded'}
            >
              {loaded ? <div>{question.question}</div> : difficulty}
              {loaded && (
                <RemoveFromCategory
                  categoryQuestionId={categoryQuestionId}
                  categoryId={categoryId}
                  gameName={gameName}
                  difficulty={difficulty}
                  questionId={this.state.questionId}
                />
              )}
            </DropTarget>
          );
        }}
      </Mutation>
    );
  }
}

export default AddToCategory;
export { ADD_TO_CATEGORY_MUTATION };
