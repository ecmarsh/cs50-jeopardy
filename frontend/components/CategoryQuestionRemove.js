import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Button from './styles/Button';
import Error from './ErrorMessage';
import { CATEGORY_QUESTIONS_QUERY } from './CategoryQuestions';
import { GAME_QUESTIONS_QUERY } from './QuestionsList';
import { GAME_SUMMARY_QUERY } from './GameSummary';
import { TextLarger } from './styles/Utilities';
import * as R from 'ramda';

const REMOVE_FROM_CATEGORY_MUTATION = gql`
  mutation REMOVE_FROM_CATEGORY_MUTATION($id: ID!) {
    removeFromCategory(id: $id) {
      id
    }
  }
`;

class RemoveFromCategory extends Component {
  static propTypes = {
    children: PropTypes.string,
    categoryQuestionId: PropTypes.string.isRequired,
  };
  static defaultProps = {
    children: `âœ•`,
  };

  handleClick = async (e, removeFromCategoryMutation) => {
    e.preventDefault();
    const {
      questionId,
      categoryQuestionId,
      categoryId,
      difficulty,
      gameName,
    } = this.props;

    const res = await removeFromCategoryMutation({
      // Mutation variables
      variables: {
        id: categoryQuestionId,
      },
      // Update local data
      update: (cache, { data: { removeFromCategory } }) => {
        // Update the question so it knows it's not in a category
        const { questions } = cache.readQuery({
          query: GAME_QUESTIONS_QUERY,
          variables: { gameName },
        });
        const questionIndex = questions.findIndex(
          question => question.id === questionId
        );

        const idx = R.findIndex(R.propEq('id', questionId));

        // Filter the category question out
        questions[questionIndex].categoryQuestions = questions[
          questionIndex
        ].categoryQuestions.filter(
          categoryQuestion => categoryQuestion.id !== removeFromCategory.id
        );

        // Write back the questions data
        cache.writeQuery({
          query: GAME_QUESTIONS_QUERY,
          variables: { gameName },
          data: { questions },
        });
        // Read, update, writeback the loaded questions cache
        const { categoryQuestions } = cache.readQuery({
          query: CATEGORY_QUESTIONS_QUERY,
          variables: { categoryId },
        });
        // Remove the question from the category
        categoryQuestions[difficulty - 1].question = null;
        // Write back the cache
        cache.writeQuery({
          query: CATEGORY_QUESTIONS_QUERY,
          variables: { categoryId },
          data: { categoryQuestions },
        });
      },
      optimisticResponse: {
        __typename: 'Mutation',
        removeFromCategory: {
          id: categoryQuestionId,
          __typename: 'CategoryQuestion',
        },
      },
    });
  };

  render() {
    const { gameName } = this.props;
    return (
      <Mutation
        mutation={REMOVE_FROM_CATEGORY_MUTATION}
        refetchQueries={[
          {
            query: GAME_SUMMARY_QUERY,
            variables: { gameName },
          },
        ]}
      >
        {(removeFromCategory, { loading, error }) => {
          if (loading) return <p>Removing..</p>;
          if (error) return <Error error={error} />;
          return (
            <Button
              close
              onClick={e => this.handleClick(e, removeFromCategory)}
            >
              <TextLarger>{this.props.children}</TextLarger>
            </Button>
          );
        }}
      </Mutation>
    );
  }
}

export default RemoveFromCategory;
export { REMOVE_FROM_CATEGORY_MUTATION };
