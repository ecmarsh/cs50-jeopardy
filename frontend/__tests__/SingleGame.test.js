import { UserGameListItem } from '../components/GamesUser';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeGame = {
  id: 'GAMEID12345',
  name: 'Awesome Game',
  createdAt: '2019-03-27T05:48:47.017Z',
};

describe('<GameListItem/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<UserGameListItem game={fakeGame} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
