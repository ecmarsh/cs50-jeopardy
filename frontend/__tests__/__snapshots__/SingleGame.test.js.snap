// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameListItem/> renders and matches the snapshot 1`] = `
<ListItem__LeftListItem
  key="GAMEID12345"
>
  <Link
    href={
      Object {
        "pathname": "/setup",
        "query": Object {
          "game": "Awesome Game",
        },
      }
    }
  >
    <a>
      <h4>
        Awesome Game
      </h4>
      <GameStatus
        gameName="Awesome Game"
      />
      <p>
        <small>
          March 26, 2019 10:48 pm
        </small>
      </p>
    </a>
  </Link>
  <DeleteGame
    gameName="Awesome Game"
  >
    <CloseSquareO />
  </DeleteGame>
</ListItem__LeftListItem>
`;
